index.rst

```diff
@@ -1,61 +1,263 @@
+
+-  Alternative architectures.
+
+   There are some existing frameworks both in industry and adopted by different observatories that, in principle, could provide a viable alternative to DDS as a middleware though they implement different architectures.
+
+   Probably the best example of frameworks on this category is `TANGO`_ which, in turn, is designed on top of the `CORBA`_ middleware.
+
+   .. note:
+
+      It is worth noting that both CORBA and DDS standards are managed by the same organization, the Object Management Group (`OMG`_`) and both rely on the Interface Description Language (`IDL`_).
+
+   Contrary to DDS, which defines a data-driven (publish-subscribe) architecture, CORBA implements an object-oriented model which is more suitable for a hierarchical system architecture.
+
+   Although it would be, in principle, possible to use CORBA in a data-driven scenario, it is not what it was designed for, which makes it hard to anticipate pitfalls we could encounter in the adoption process.
+
+   Therefore, even though we explored some of these alternative architectures systems, and some of them shows some promise, it seems like a larger risk than to find a suitable publish-subscribe alternative to DDS.
+
+.. _TANGO: https://www.esrf.fr/computing/cs/tango/tango_doc/icaleps99/WA2I01.html
+.. _CORBA: https://www.corba.org
+.. _OMG: https://www.omg.org
+.. _IDL: https://www.omg.org/spec/IDL/4.2/About-IDL/
+
+After extensively researching alternatives to ADLink-OpenSpliceDDS (and DDS) we have finally converged into a best alternative; `Kafka`_.
+
+`Kafka`_ is an open source event streaming platform that is broadly used in industry.
+In fact, it is already an integral part of the Rubin-OCS, as it is used in the EFD to transport the data from DDS to influxDB (:sqr:`034` :cite:`SQR-034`).
+
+.. _Kafka: https://kafka.apache.org
+
+The fact that we are already using Kafka in the system reliably to ingest data into the EFD gives us confidence that it is, at the very least, able to handle the overall data throughput.
+Our main concern is than to verify that Kafka can handle the latency requirements of our system.
+In principle, Kafka is advertised as a "real-time" system and numerous benchmarks exists online showing it can reach latencies at the millisecond regime.
+Nevertheless, it is unclear those benchmarks would be applied to our systems constrains, giving the typical message size, network architecture and other relevant factors.
+
+We then proceeded to perform benchmarks with the intention to evaluate Kafka's performance considering our system architecture.
+The results, which are detailed in :tstn:`033`, are encouraging.
+In summary, we obtain similar latency levels for both Kafka and DDS.
+In terms of throughput, DDS is considerably better than Kafka for smaller messages, though we obtain similar values for larger messages.
```

Comment:
More importantly, the throughput is good enough for Kafka  (the EFD can keep up with our control system). You mention this early on, and I would encourage you to say it again here.